repository/httpRemoteRepository.go

```diff
@@ -0,0 +1,296 @@
+		client:              client,
+		urlSchema:           urlSchema,
+		useCache:            true,
+		remoteConfiguration: remoteConfig,
+	}, nil
+}
+
+// CreateHTTPClient creates a httpClient for the given remote settings.
+func CreateHTTPClient(config *core.Remote) (*http.Client, error) {
+	httpClient := &http.Client{
+		Timeout: 10 * time.Second,
+	}
+
+	transport, err := createProxyHTTPTransport(config)
+	if err != nil {
+		return nil, err
+	}
+	httpClient.Transport = transport
+
+	return httpClient, nil
+}
+
+func createProxyHTTPTransport(config *core.Remote) (*http.Transport, error) {
+	transport := &http.Transport{}
+
+	if config.ProxySettings.Enabled {
+		proxyURLString := config.ProxySettings.CreateURL()
+		core.GetLogger().Infof("configure http client to use proxy %s", proxyURLString)
+
+		proxyURL, err := neturl.Parse(proxyURLString)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse proxy url %s: %w", proxyURLString, err)
+		}
+		transport.Proxy = http.ProxyURL(proxyURL)
+		appendProxyAuthorizationIfRequired(transport, &config.ProxySettings)
+	}
+
+	transport.TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: config.Insecure,
+	}
+
+	return transport, nil
+}
+
+func appendProxyAuthorizationIfRequired(transport *http.Transport, proxySettings *core.ProxySettings) {
+	if proxySettings.Username != "" {
+		authorization := proxySettings.Username + ":" + proxySettings.Password
+		basicAuthorization := "Basic " + base64.StdEncoding.EncodeToString([]byte(authorization))
+		if transport.ProxyConnectHeader == nil {
+			transport.ProxyConnectHeader = make(map[string][]string)
+		}
+
+		transport.ProxyConnectHeader.Add("Proxy-Authorization", basicAuthorization)
+	}
+}
+
+// GetLatest returns the detail about the latest dogu from the remote server by name.
+func (r *httpRemote) GetLatest(_ context.Context, name dogu.QualifiedName) (*core.Dogu, error) {
+	err := name.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("qualified dogu name is not valid (name: %s): %w", name.String(), err)
+	}
+	requestUrl := r.urlSchema.Get(name.String())
+	cacheDirectory := filepath.Join(r.endpointCacheDir, name.String())
+	return r.receiveDoguFromRemoteOrCache(requestUrl, cacheDirectory)
+}
+
+// Get returns a version specific detail about the dogu.
+func (r *httpRemote) Get(_ context.Context, doguVersion dogu.QualifiedVersion) (*core.Dogu, error) {
+	err := doguVersion.Name.Validate()
+	if err != nil {
+		return nil, fmt.Errorf("qualified dogu name is not valid (name: %s): %w", doguVersion.Name.String(), err)
+	}
+	requestUrl := r.urlSchema.GetVersion(doguVersion.Name.String(), doguVersion.Version.Raw)
+	cacheDirectory := filepath.Join(r.endpointCacheDir, doguVersion.Name.String(), doguVersion.Version.Raw)
+	return r.receiveDoguFromRemoteOrCache(requestUrl, cacheDirectory)
+}
+
+func (r *httpRemote) receiveDoguFromRemoteOrCache(requestUrl string, dirname string) (*core.Dogu, error) {
+	var remoteDogu, err = r.readCachedDogu(dirname)
```

Comment:
You are loosing the initial cache error in the retry. Rename the error and wrap it.